### DISCUSSION TITLE
Idea: bundle a second painting app - for more painterly styles

### DISCUSSION THREAD

**Discussion by blurymind** [4ian/GDevelop#2329] [Enhancement ideas (and their technical discussions)]
Just putting this out here to see if there is interest :)

So basically piskel is awesome and fits our needs because it can do animated sprites, tiling sprites and specializes as a pixel editor- perfect for all retro looking games! So as a pixel editor it  is probably our best choice!

BUT

It's not great if you want to do any other style of art than pixelart. It reduces colors and alpha blending when you import images in it. It doesnt have any non-pixelart brushes or tools.

Sometimes people like to have the option to do a more painterly art in their game - for backgrounds for example. This is where they cant use piskel.
But fear not - I kind of found a cool app that is open source and html5 that can fill that gap.

https://github.com/thenickdude/chickenpaint

Pros:
- awesome watercolor brush and many different brush blending modes, layer effects and much more
- supports tablet pressure sensitivity!
- already popular among online art board communities, as it is a port of chibipaint
- doesn't degrade quality of images you edit - doesnt reduce their color palette or try to make them  more pixel arty
- has lots of tools and intuitive ui
- you can do higher resolution styles that are not pixelated - directly in the browser

Cons:
- 5mb distributable size
- it can not do animation (but maybe some day in the future it could?) 
- not a good choice for pixel art, because it can not keep a low color palette like piskel

My proposition is to bundle it and add it to the menu of standalone images - so people can also do the painterly style for backgrounds. Sure it will add 5mb, but what is 5mb for such a full featured painting app

**Comment by Bouh** [comment]
It is a very nice application, however I would advise users to use external software to have more advanced functions than any integrated editor.

Integrated a new drawing editor does not seem to me to be useful.

Having Piskel is good because it gives you small basics and it's nice for prototyping. (And on every computer there is a basic drawing software. so ..)

There is no community request for another drawing editor.
It seems like a waste of time for not much, even if the application is powerfull.

You have to be careful not to include a whole bunch of editors in GDevelop,
GDevelop is a game engine above all.

That's why I would advise people to use an external editor (krita,gimp,paint.net,photoshop,etc...)

Let the user use the drawing software he wants, without cluttering it with an editor he will not use.

**Comment by blurymind** [comment]
yes I do agree on those points, but there was actually someone complaining about piskel removing color information from their imported images that they tried to edit. Piskel should never ever be used on editing non pixel art images imported from another painting app - but chickenpaint would be ok to do so and keep the fidelity.

krita, gimp  and the others are big apps that take much more space than 5mb, eventhough they do offer some advance features. Saying that though, I have to insist that even for its size this is an impressively powerful app. It has a lot of krita's features - layer masking, effects, transorm tool with rotate (piskel cant do rotation) - it has a hell of a lot of features compared. It kind of sucks that one has to compile and run it to try them in order to see - as the author hasn't made a proper online demo that doesnt require registering at some oekaki board.

I was impressed enough to make a post about it here though :)  in the case of there being more interest in this - it wouldnt take that much effort to add

**Comment by ddabrahim** [comment]
To be honest I don't like the way Piskel integrate with GDevelop.
In my opinion it should integrate better before we have a second app/feature for making other style of art.

What I dislike:
theme looks different than the theme of GD, at least the base colors should be copied
Piskel specific buttons, we don't need them.
The most disturbing is that we have 2 save buttons and an export button to do the same thing essentially. They should be removed and have buttons for GD use only.
If anyone wants to use Piskel outside GD can download it separately from their official page
GD specific buttons looks out of place, some styling would be nice.
No "Save as" button for GD and when you click save, no question if you want to replace a current image.

Having Piskel or any other app integrated in to GD is awesome and really appreciate it but having so many things out of place makes GD look and feel poor quality.
This is also apply to the new dialog editor. It is awesome, and very useful huge thanks, but the look and feel is just out of place.

We need these 3rd party editors to be better integrated first and then add more otherwise GD can never leave beta in my opinion.

**Comment by blurymind** [comment]
The theme thing is harder to do than it looks, because GD has different themes too, applying a style to an external app has to be maintained for each theme. I am not going to invest any time in trying that, but if you want to you are welcome to do a PR =)

The save functionality is actually different. The one in piskel let's you save as any file. The one in GD is saving the pixels as object metadata. What you are asking me to do is to remove functionality here, so users can't export piskel files when using it in GD. This is again something I have no interest in investing any time doing as it has no gains in it. To be fair the save button in piskel is kind of hard to find to begin with, so I am kind of confused why it bothers you or even makes you think it does the same as save to gd button 😂

You can now hide the entire save to GD bar if you like, so you are welcome to do that too if you like. But again using piskels save to will save it as a piskel file, not as something GD can directly use

It's worth remembering that using any external apps in GD will always have some integration limitations. But I think that is ok. The user should know these apps are not developed by the GD devs. We should never mislead users they are a part of our code base, otherwise we will start getting feature requests and bug reports on them too here.

Finally this issue post here is not about piskel, and I will not be doing any piskel requests here, so please stay on topic. You are kind of going all over the place on new things in GD which are not exactly the way you want them to. What new dialog editor? Do you mean yarn?

**Comment by ddabrahim** [comment]
>I am kind of confused why it bothers you or even makes you think it does the same as save to gd button 😂
But I think that is ok. The user should know these apps are not developed by the GD devs

It does not look polished, that's all. I don't personally mind it, I make games with a bucket if helps and get me where I want to be. But I do have legit concerns regarding the future of GD. Quality of life, look and feel that sort of things are also important if you want to advertise something and make it popular. If it looks like *** and feels like *** even though it works and get the job done it might makes some people to reconsider it if they can and should rely their business on this thing or not.

If we want to be honest, the growing popularity of GD is largely effected by the fact Construct become a pay monthly web only service. If Construct 3 would be released as an offline, desktop app and the price would be very friendly, it could kill GD instantly, any moment simply because of the overall look and feel of Construct is more polished. 

Don't get me wrong, I do appreciate all the work contributed to GD and a new editor again sounds awesome but I think GD reached the point where devs should honestly considered focusing on quality over quantity.

But it is just my opinion, I'm not trying to tell anyone what they supposed to do and I am not saying you or anyone doing a bad job, but only that it does not look and feel polished and I simply don't like the overall look and feel of GDevelop in particular because of the integrated editors. That's all. 👍

**Comment by blurymind** [comment]
While construct and gd have the event sheet in common, I'd like to think of them as different beasts.

The construct guys are running a business and while their product is becoming more and more different than gd. Both  have pros and cons.

Construct pros:
- looks more polished
- has active developers working on it, delivering new features
cons:
- costs an arm and a leg for what it delivers
- you are renting it, never owning
- closed source, so you can't change a thing

Gdevelop pros:
- Free
- It's open source and anyone who puts enough time can change things they don't like ( including @ddabrahim ) - truly owned by its community
- Does have active devs - the community. Thats why the GD community will get what they want more than construct's community
- can take bolder choices, such as including more  powerful editors than construct - out of the box.
- the IDE can take advantage of the best from the open source world in terms of webapps. That sort of thing is a much harder choice for the for-profit construct3 guys who need to constantly prove their buck for value and keep pro appearance

cons:
- the development  is all done in contribution, as hobby
- It does not look as polished as construct3 - as a singular app
- It's always a work in progress, but is constantly improving

The strength of open source is in its ecosystem - or the way different open source projects come together to complement each other and create something new and more powerful.

Now correct me if I'm wrong, but construct3's built in sprite editor does not have the ability to store layer data :p also its sprite editing tools are pretty darned basic and limited.
To make things even crappier - none of the users can improve them. In contrast - gd uses piskel, which is open source and has a git tracker. You want to add a tool? You can.

Before choosing GD over construct, you have to accept its pros and cons. What are you willing to take? More tools and layer data or a more unified theme?

I am willing to bet that gamedevs dont care about what color a button is as much as getting their game finished. If we give them better tools to get there, they will love gd more than constrcut.

I  bet that you are using the default light theme and thats why piskel looks so different. I use the dark theme. :)

**Comment by ddabrahim** [comment]
>What are you willing to take? More tools and layer data or a more unified theme?

You have a point there and I also give you that being free and open-source under MIT license is a huge advantage to those who interested in developing, maintaining their own in-house version and able to do whatever they want with it.

If need to choose, yes I would choose the ugly open-source one with more tools :D
But a unified theme is also important imo if you think commercial. The best would be to have both :)

Not sure why I can't see GD being a completely free tool forever. Even though it is great and free and open, I don't see that much attention around it from other devs. Like Godot for example was a breaking news since day 1 and just keep growing, it is insane. I can see Godot dominate the world in few years but I just don't see that GD coming anywhere near that popularity just because it is free and open and have lot of tools. So at some point GD may need to switch track and a unified, polished theme and smooth user experience will be a must. 
But I guess I'm going way too far in to the future and some may don't even like the direction I am going with this so better I don't push it :P 

>I bet that you are using the default light theme and thats why piskel looks so different.

Yes, that is the case :D 
Maybe I should try the dark theme sometime, I hope ChickenPaint do have a dark theme 👍

**Comment by 4ian** [comment]
You both have good points :) Let's not go too much into comparisons and keep the thread about the potential painting app, but a few things though:

- Construct is definitely a great app and game engine. GDevelop and C3 have a slightly different philosophy behind them and how they are developed and while similar, they might also differ in the future.
- Godot is also a great engine but more targeted at something different (3D/high performance) and with not the same highlight put on the learning curve/visual programming side. (though this can change, and I also mentioned at some point that GDevelop could export to Godot one day).
- GDevelop popularity is less significant than these game engines, but is still growing every month and is definitely in range comparable to other game engines.

Being an "outsider" is also a position where I (and contributors) can actually try new things, make bold decisions as blurymind hinted, trying things that can actually be out of reach of other game engines. Question of philosophy and choices - which is a reason why it's always good to have alternatives in terms of service/product/technology/whatever (browser, front-end frameworks, game engines, supermarket, bakeries, you see my point).

> But a unified theme is also important imo if you think commercial. The best would be to have both :)

I do agree, I think things are even going further in terms of polishing needed. For example, I'm regularly fixing things on the wiki because people want to contribute but don't do always it with the level of quality that is necessary for a software like GDevelop. Documentation is important and hard to get right.

> Not sure why I can't see GD being a completely free tool forever.

This is surely because it's actually *too good* for an open source product ;) GDevelop is too good for what people give for it.
Joke aside, certainly still lacking polishing and a good marketing team behind it, but we're at a good level already. Still trying to find proper ways to make this sustainable in the very long term.

**Comment by blurymind** [comment]
> Yes, that is the case :D
> Maybe I should try the dark theme sometime, I hope ChickenPaint do have a dark theme

It has a light theme actually 
![screenshot](https://user-images.githubusercontent.com/6495061/68548457-10a71180-03e5-11ea-9fdd-28f1bdcfddae.png)

I left a few comments at its tracker and was happy to find that the developer is actually quite active on it and responds very quickly to community input. It is a really nice painting app. Don't let those few  missing icons mislead you to think it unpolished. I raised an issue about the missing icons btw. It appears to be using gimp's icon set
